ovs/dal/datalist.py

```diff
@@ -103,16 +116,132 @@ def guids(self):
             self._execute_query()
         return self._guids
 
+    def set_key(self, key=None):
+        """
+        Sets the caching key
+        Won't override the key when a key was giving on initializing
+        :param key: Key to explicitly use
+        :type key: str
+        :return: None
+        :rtype: NoneType
+        """
+        if key is not None:
+            self._key = '{0}_{1}'.format(DataList.NAMESPACE, key)
+            self._provided_key = True
+            # Unsure whether or not the same query would apply
+            self._volatile.delete(self._key)
+            return
+        elif self._provided_key is False:
+            identifier = copy.deepcopy(self._query)
+            identifier['object'] = self._object_type.__name__
+            identifier['guids'] = 'None' if self._provided_guids is None else ','.join(self._provided_guids)
```

Comment:
I'd order the guids, because a different order of the same guids should return the same result